.github/workflows/main.yml

```diff
@@ -209,10 +270,6 @@ jobs:
           name: ${{ matrix.asset }}.${{ matrix.rid }}
           path: ${{ matrix.app }}/
 
-      - name: Set permissions
-        if: ${{ !startsWith(matrix.rid, 'win-') }}
-        run: chmod +x ${{ matrix.app }}/${{ matrix.asset }}
-
```

Comment:
Last time, I remember looking into this and found that GitHub Actions artifacts don't retain file permissions. It was a while ago so maybe that changed. Can you confirm that the downloaded release assets will have the correct permissions assigned to them?

Reply:
+x permission preserved
<img width="837" alt="image" src="https://github.com/user-attachments/assets/b8311911-f57a-419a-9ff6-ee0de5d3cb06" />